modules/web-console/web-agent/src/main/java/org/apache/ignite/console/agent/handlers/WebSocketRouter.java

```diff
@@ -18,238 +18,266 @@
+                log.trace("Received ping message [socket=" + ses + ", msg=" + frame + "]");
+
+            try {
+                ses.getRemote().sendPong(PONG_MSG);
+            }
+            catch (Throwable e) {
+                log.error("Failed to send pong to: " + ses, e);
+            }
+        }
+    }
+
+    /**
+     * @param e Error.
      */
-    @SuppressWarnings("unused")
-    private void connect(long tid) {
-        client.websocket(conOpts,
-            ws -> {
-                log.info("Connected to server: " + ws.remoteAddress());
-
-                register(ws, Addresses.SCHEMA_IMPORT_DRIVERS);
-                register(ws, Addresses.SCHEMA_IMPORT_SCHEMAS);
-                register(ws, Addresses.SCHEMA_IMPORT_METADATA);
-                register(ws, Addresses.NODE_REST);
-                register(ws, Addresses.NODE_VISOR);
-
-                MessageConsumer c1 = vertx.eventBus().consumer(Addresses.CLUSTER_TOPOLOGY, msg -> {
-                    JsonObject json = new JsonObject()
-                        .put("agentId", AGENT_ID)
-                        .put("top", msg.body());
-
-                    send(ws, Addresses.CLUSTER_TOPOLOGY, json);
-                });
-
-                ws.handler(data -> {
-                    JsonObject json = data.toJsonObject();
-
-                    String addr = json.getString("address");
-
-                    if (F.isEmpty(addr))
-                        log.error("Unexpected request: " + json);
-                    else
-                        vertx.eventBus().send(
-                            addr,
-                            json.getJsonObject("body"),
-                            asyncRes -> {
-                                Object body = asyncRes.succeeded()
-                                    ? asyncRes.result().body()
-                                    : new JsonObject().put("err", asyncRes.cause().getMessage());
-
-                                send(ws, json.getString("replyAddress"), body);
-                            });
-
-                    // log.info("Received data " + data.toString());
-                });
-
-                long pingTimer = vertx.setPeriodic(3000, v -> write(ws, PING));
-
-                ws.closeHandler(p -> {
-                    log.warning("Connection closed: " + ws.remoteAddress());
-
-                    vertx.cancelTimer(pingTimer);
-                    c1.unregister();
-
-                    curTimer = vertx.setTimer(1, this::connect);
-                });
-
-                ws.exceptionHandler(e -> log.error("EXCEPTION ON SOCKET", e));
-                // send(ws, "agent:id", AGENT_ID);
-            },
-            e -> {
-                LT.warn(log, e.getMessage());
-
-                curTimer = vertx.setTimer(3000, this::connect);
-            });
+    @OnWebSocketError
+    public void onError(Throwable e) {
+        log.error("ERR on websocket", e);
+
+        try {
+            Thread.sleep(3000);
```

Comment:
Possible this timeout should be logarithmic or zero on first reconnect.

Reply:
Fixed.